components/ai_chat/content/browser/ai_chat_tab_helper.cc

```diff
@@ -321,52 +278,105 @@ void AIChatTabHelper::GetPageContent(GetPageContentCallback callback,
     // Ocred but `remaining_page_count_` was set to total pdf page count. Hence
     // status IDS_PDF_OCR_COMPLETED or IDS_PDF_OCR_NO_RESULT won't be set.
     CheckPDFA11yTree();
-  } else {
-    if (base::Contains(kPrintPreviewRetrievalHosts,
-                       GetPageURL().host_piece())) {
-      pending_print_preview_requests_[pending_navigation_id_] =
-          std::move(callback);
-      if (is_page_loaded_) {
-        NotifyPrintPreviewRequested(false);
-      }
+    return;
+  }
+  if (base::Contains(kPrintPreviewRetrievalHosts, GetPageURL().host_piece()) &&
+      print_preview_extraction_delegate_ != nullptr) {
+    if (is_page_loaded_) {
+      // Get content using a printing / OCR mechanism, instead of
+      // directly from the source.
+      print_preview_extraction_delegate_->Extract(
+          is_pdf,
+          base::BindOnce(&AIChatTabHelper::OnExtractPrintPreviewContentComplete,
+                         weak_ptr_factory_.GetWeakPtr(), std::move(callback)));
     } else {
-      FetchPageContent(web_contents(), invalidation_token, std::move(callback));
+      // Run print preview extraction after load
+      if (pending_get_page_content_callback_) {
+        std::move(pending_get_page_content_callback_).Run("", false, "");
+      }
+      pending_get_page_content_callback_ = std::move(callback);
```

Comment:
I think this can be pull into something like `MaybePrintPreviewExact(...)` and use it here and `OnFetchPageContentComplete(...)`

Reply:
This is actually doing something slightly different to `OnFetchPageContentComplete` which is attempting to re-run regular `FetchPageContent` after page load, not just print preview extractor. So I've made 2 reusable functions to cover it.